core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -304,12 +304,16 @@ private void reportMetrics(
       MetricsReport report,
       Supplier<Map<String, String>> headers) {
     reporter.report(report);
-    client.post(
-        paths.metrics(tableIdentifier),
-        ReportMetricsRequest.of(report),
-        null,
-        headers,
-        ErrorHandlers.defaultErrorHandler());
+    try {
+      client.post(
+          paths.metrics(tableIdentifier),
+          ReportMetricsRequest.of(report),
+          null,
+          headers,
+          ErrorHandlers.defaultErrorHandler());
+    } catch (Exception e) {
+      LOG.warn("Failed to report metrics to REST endpoint for table {}", tableIdentifier, e);
```

Comment:
nit: Print out the details of the report?

Reply:
it's being printed to the log right outside the try-catch block, so not sure we'd want to repeat the report details in this warning message